lib/dspam/CMakeLists.txt

```diff
@@ -0,0 +1,116 @@
+#################################################################################
+# CMakeLists.txt for DSPAM Library
+#################################################################################
+# This file builds the DSPAM spam filtering library which provides statistical
+# spam detection and filtering capabilities.
+
+# Library sources
+set(DSPAM_SOURCES
+    src/base64.c
+    src/bnr.c
+    src/buffer.c
+    src/config_shared.c
+    src/decode.c
+    src/diction.c
+    src/error.c
+    src/hash.c
+    src/hash_drv.c
+    src/heap.c
+    src/libdspam.c
+    src/list.c
+    src/nodetree.c
+    src/pref.c
+    src/read_config.c
+    src/tokenizer.c
+    src/util.c
+)
+
+# Create the DSPAM library
+add_library(mahogany_dspam ${DSPAM_SOURCES})
+
+# Set library properties
+set_target_properties(mahogany_dspam PROPERTIES
+    OUTPUT_NAME dspam
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR}
+)
+
+# Include directories
+target_include_directories(mahogany_dspam
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
+        $<INSTALL_INTERFACE:include/mahogany/dspam>
```

Comment:
We don't need to distinguish between build and install interfaces, this library is not going to be installed, it is only built as part of Mahogany.